### DISCUSSION TITLE
How this tool can be utilized for Power Platform ALM?

### DISCUSSION THREAD

**Discussion by tomsinsight** [getporter/porter#1912] [Q&A]
Hello Team,

It seems this is more useful for the Azure artifacts ALM - Can it be utilized for the Power Platform Solution Artifacts as well?

Thank you very much for your time in advance!

Best,

**Comment by squillace** [comment]
Hi @tomsinsight, sure, it can be used for pretty much anything. Try out the https://github.com/squillace/porter-pac mixin. Lemme know how it goes.

**Reply by tomsinsight** [reply]
Thank you for sharing the information, is there any step by step process of using Porter Pac with Power Platform? for example, What this does exactly?

Does it import power platform solution from one environment to another?
does it check in checkout devops repository of Power Platform solution? 
do we have to create the repository on our own or porter does it?
Is it possible through commands/ built in app?
How does it managed security wise? only admin can perform or others too?

If you could help me with any reference link that shares step by step process/guidance of using Porter Pac with Power platform ALM - would be highly appreciated.

**Comment by squillace** [comment]
Hi @tomsinsight, this file shows what the mixin does: https://github.com/squillace/porter-pac/blob/main/pkg/pac/build.go. It installs the PowerApp CLI from nuget: https://github.com/squillace/porter-pac/blob/f06277c20ca310daf763975643c3eacd8df43b10/pkg/pac/build.go#L60. The nuget package is this one: https://www.nuget.org/packages/Microsoft.PowerApps.CLI.Core.linux-x64/. So the docs are https://docs.microsoft.com/en-us/powerapps/developer/component-framework/create-custom-controls-using-pcf. 

As a result, you're going to have the standard exec-style commands available to you in the porter.yaml file. An example is here: https://github.com/squillace/hello-pac/blob/main/porter.yaml. Lemme know if this doesn't get you started. Technically, I'm on vacation today, but hey, cool things are always fun. Tag me back here if you want to discuss it more.

**Comment by squillace** [comment]
and I've written basic mixins for powerbi (https://github.com/squillace/porter-pbicli) with an example for that (https://github.com/squillace/hello-powerbi/blob/main/porter.yaml). You can make porter and cnab automate pretty much anything and have that pipeline remain immutable like a good container image should.
